modules/virt-configuring-cluster-dpdk.adoc

```diff
@@ -0,0 +1,117 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/vm_networking/virt-attaching-vm-to-sriov-network.adoc
+
+:_content-type: PROCEDURE
+[id="virt-configuring-cluster-dpdk_{context}"]
+= Configuring a cluster for DPDK workloads
+
+Use this procedure to configure an {product-title} cluster to run Data Plane Development Kit (DPDK) workloads.
+
+:FeatureName: Configuring a cluster for DPDK workloads
+include::snippets/technology-preview.adoc[]
+
+.Prerequisites
+* You have access to the cluster as a user with `cluster-admin` permissions.
+* You have installed the OpenShift CLI (`oc`).
+* You have installed the SR-IOV Network Operator.
+* You have installed the Node Tuning Operator.
+
+.Procedure 
+. Map your compute nodes topology to determine which non-uniform memory access (NUMA) CPUs will be isolated for DPDK applications and which ones will be reserved for the OS.
+. Label a subset of the compute nodes with a custom role, for example, `worker-dpdk`:
++
+[source,terminal]
+----
+$ oc label node <node_name> node-role.kubernetes.io/worker-dpdk=""
+---- 
+
+. Create a new `MachineConfigPool` manifest that contains the `worker-dpdk` label in the `spec.machineConfigSelector` object:
++
+.Example `MachineConfigPool` manifest
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: dpdk-workers
+  labels:
+    machineconfiguration.openshift.io/role: dpdk-workers
+    pools.operator.machineconfiguration.openshift.io/dpdk-workers: ""
+spec:
+  configuration:
+  machineConfigSelector:
+    matchExpressions:
+      - key: machineconfiguration.openshift.io/role
+        operator: In
+        values:
+          - worker
+          - worker-dpdk
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker-dpdk: ""
+----
+
+. Generate a performance profile by using the Performance Profile Creator (PPC) command-line tool, which is part of the Node Tuning Operator. The PPC requires the following arguments:
+** `mcp-name`: The machine config pool name corresponding to the target machines. 
+** `reserved-cpu-count`: The number of reserved CPUs for house keeping. 
+** `rt-kernel`: Enable real-time kernel.
++
+.Example `PerformanceProfile` manifest
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+  name: profile-1
+spec:
+  cpu:
+    isolated: 2-39,42-79
+    reserved: 0,1,40,41
+  hugepages:
+    defaultHugepagesSize: "1G"
+    pages:
+      - count: 32
+        size: "1G"
+  numa:
+    topologyPolicy: "single-numa-node"
+  nodeSelector:
+    node-role.kubernetes.io/worker-dpdk: "" 
+----
+
```

Comment:
After applying these two manifests, there should be a gradual reboot of the relevant nodes.
Please consider adding a note about it.

Reply:
Does the user initiate the node reboot or does it happen automatically when the MCP and PP manifests are applied?

Reply:
@orelmisan Does this sound okay: 

> Perform a graceful restart of the compute nodes after you apply the `MachineConfigPool` and `PerformanceProfile`manifests.

I can also link to this procedure if it is relevant: https://docs.openshift.com/container-platform/4.9/nodes/nodes/nodes-nodes-rebooting.html#nodes-nodes-rebooting-gracefully_nodes-nodes-rebooting

Reply:
> Does the user initiate the node reboot or does it happen automatically when the MCP and PP manifests are applied?

The nodes are automatically rebooted.
I don't know if the user has the ability to manually control this process.